services/web/project/app/views.py

```diff
@@ -35,48 +41,83 @@ def new_question():
+        return redirect(url_for(
+            "views.question",
+            question_id=q.id))
 
     return render_template("new_question.html", form=form)
 
 
 @views.route('/questions/<question_id>', methods=['GET', 'POST'])
 def question(question_id):
-    # Get question from DB
-    question = Question.query.get(question_id)
-    # Validate that question exists; if not, route to questions forum
-    if question is None:
-        questions()
-        questions()
-        questions()
-    # Get list of answers for question
-    question.answers = Answer.query.filter_by(questionId=question.id).all()
-    question.numAnswers = len(question.answers)
-    # Get the question's creator and assign it as an attribute
-    question.creator = User.query.get(question.userId)
-
-    # Process form
-    form = NewAnswerForm()
-    if form.validate_on_submit():
-        body = form.body.data
-        # Add answer to DB
-        a = Answer(body, current_user.id, question.id)
-        db.session.add(a)
-        db.session.commit()
-        return redirect(url_for('views.question', question_id=question_id))
+    result = QuestionCtrl.getQuestion(question_id)
+    # The controller returns the results as a list:
+    # result[0] : Operation Status => value = "SUCCESS" | "ERROR"
+    # result[1] : Message => value "succes_or_eror_message"
+    # result[2] : Question object
+    #   associated wit result[0] = "SUCCESS" and result[1] = "QUESTION_FOUND"
+    # result[3] : Form object
+    #   associated wit result[0] = "SUCCESS" and result[1] = "QUESTION_FOUND"
 
-    # For each answer, add the creator as an attribute
-    for a in question.answers:
-        a.creator = User.query.get(a.userId)
-    return render_template('question.html', question=question, form=form)
+    # If operation was executed successfully
+    if(result[0] == "SUCCESS"):
+        message = result[1]
+        # GET request received to retrieve questio from DB
+        if(message == "QUESTION_FOUND"):
+            _question = result[2]
+            _form = result[3]
+            return render_template(
+                "question.html",
+                question=_question,
+                form=_form)
+        # else message = "NEW_ANSWER_CREATED"
+        # POST request received on new answer form submission
+        else:
+            _question_id = result[2]
+            return redirect(url_for(
+                "views.question",
+                question_id=_question_id))
+    # else result[0] = "ERROR"
+    else:
+        # Redirect to questions forum
+        return redirect(url_for("views.questions"))
 
-# BUG: The function below does not return the files in static as expected.
-# @views.route("/static/<path:filename>")
-# def static_file(filename):
-#     return send_from_directory(app.config["STATIC_FOLDER"], filename)
+
+# Accept Answer
+@views.route('/questions/accept_answer/<answer_id>/<question_id>', methods=['GET'])
```

Comment:
I approve, but in the future we need to go `/questions/<question_id>/accept_answer/<answer_id>` makes more sense